automation-roles/30-provision-infra/provision-aws/tasks/provision-ipi.yml

```diff
@@ -0,0 +1,38 @@
+---
+- set_fact:
+    _openshift_config: "{{ all_config.openshift | json_query(query) | first | default({}) }}"
+  vars:
+    query: >-
+      [?name=='{{ _current_openshift_cluster.name }}']
+
+- name: Check if OpenShift cluster already exists by checking kubeadmin password secret
+  include_role: 
+    name: vault-get-secret
+  vars:
+    secret_name: "{{ _current_openshift_cluster.name }}-kubeadmin-password"
+    secret_group: "{{ environment_name }}"
+
+- set_fact:
+    _kubeadmin_password: "{{ secret_value }}"
+
+# Run OpenShift IPI installer if no OpenShift cluster does not exist yet
+- block:
+  - name: Run IPI installer for OpenShift cluster {{ _current_openshift_cluster.name }}
```

Comment:
Make sure you send the output of this long-running task to a log file, something like this:
```
- name: Create ROSA cluster, logs can be found in {{ status_dir }}/log/{{ _current_openshift_cluster.name }}-create-cluster.log
  shell: |
    {{ lookup('template','rosa-create-cluster.j2') }} \
      >> {{ status_dir }}/log/{{ _current_openshift_cluster.name }}-create-cluster.log 2>&1
  environment:
    AWS_ACCESS_KEY_ID: "{{ _aws_access_key }}"
    AWS_SECRET_ACCESS_KEY: "{{ _aws_secret_access_key }}"
  when: _rosa_describe_cluster.rc != 0
```

Reply:
```
  - name: Run IPI installer for OpenShift cluster {{ _current_openshift_cluster.name }}
    shell:
      openshift-install create cluster --dir={{ status_dir }}/aws-ipi/{{ _current_openshift_cluster.name }} >> {{ status_dir }}/log/{{ _current_openshift_cluster.name }}-create-cluster.log 2>&1
    environment:
      AWS_ACCESS_KEY_ID: "{{ _aws_access_key }}"
      AWS_SECRET_ACCESS_KEY: "{{ _aws_secret_access_key }}"
```